File: app/src/main/java/com/example/myapp/FragmentSettings.java
package com.example.myapp;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ScrollView;

import com.google.android.material.checkbox.MaterialCheckBox;
import com.google.android.material.radiobutton.MaterialRadioButton;

public class FragmentSettings extends Fragment {

    ScrollView scrollView;
    MaterialCheckBox checkBoxPressure;
    MaterialCheckBox checkBoxWind;
    MaterialCheckBox checkBoxSunMoving;
    MaterialCheckBox checkBoxHumidity;
    MaterialRadioButton rbThemeSystem;
    MaterialRadioButton rbThemeLight;
    MaterialRadioButton rbThemeDark;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_settings, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        FragmentActivity fa = getActivity();
        if (fa != null) {
            fa.setTitle(R.string.label_settings);
        }
        findViewsById(view);
        restoreViewsValueFromSettingsContainer();
        initListenerForCheckButtons();
        initListenerForThemesButtons();
    }

    private void findViewsById(View view) {
        scrollView = view.findViewById(R.id.scrollView);
        checkBoxPressure = view.findViewById(R.id.checkBoxPressure);
        checkBoxWind = view.findViewById(R.id.checkBoxWind);
        checkBoxSunMoving = view.findViewById(R.id.checkBoxSunMoving);
        checkBoxHumidity = view.findViewById(R.id.checkBoxHumidity);
        rbThemeSystem = view.findViewById(R.id.rbThemeSystem);
        rbThemeLight = view.findViewById(R.id.rbThemeLight);
        rbThemeDark = view.findViewById(R.id.rbThemeDark);
    }

    private void initListenerForCheckButtons() {

        View.OnClickListener chkClickListener = view -> {
            SettingsContainer sc = SettingsContainer.getInstance();
            sc.isChkBoxPressure = checkBoxPressure.isChecked();
            sc.isChkBoxWind = checkBoxWind.isChecked();
            sc.isChkBoxSunMoving = checkBoxSunMoving.isChecked();
            sc.isChkBoxHumidity = checkBoxHumidity.isChecked();
        };

        checkBoxHumidity.setOnClickListener(chkClickListener);
        checkBoxPressure.setOnClickListener(chkClickListener);
        checkBoxSunMoving.setOnClickListener(chkClickListener);
        checkBoxWind.setOnClickListener(chkClickListener);
    }

    private void initListenerForThemesButtons() {

        View.OnClickListener rbClickListener = view -> {
            SettingsContainer sc = SettingsContainer.getInstance();
            sc.isThemeSystem = rbThemeSystem.isChecked();
            sc.isThemeLight = rbThemeLight.isChecked();
            sc.isThemeDark = rbThemeDark.isChecked();
            FragmentActivity fa = getActivity();
            if (fa != null) {
                fa.recreate();
            }
        };

        rbThemeSystem.setOnClickListener(rbClickListener);
        rbThemeLight.setOnClickListener(rbClickListener);
        rbThemeDark.setOnClickListener(rbClickListener);
    }

    private void restoreViewsValueFromSettingsContainer() {
        SettingsContainer sc = SettingsContainer.getInstance();
        checkBoxPressure.setChecked(sc.isChkBoxPressure);
        checkBoxWind.setChecked(sc.isChkBoxWind);
        checkBoxSunMoving.setChecked(sc.isChkBoxSunMoving);
        checkBoxHumidity.setChecked(sc.isChkBoxHumidity);
        rbThemeSystem.setChecked(sc.isThemeSystem);
        rbThemeLight.setChecked(sc.isThemeLight);
        rbThemeDark.setChecked(sc.isThemeDark);
    }
}

File: app/src/main/java/com/chplalex/shaman/ui/fragment/FragmentSettings.kt
package com.chplalex.shaman.ui.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.Menu
import android.view.MenuInflater
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup
import android.widget.RadioGroup
import com.chplalex.shaman.R
import com.chplalex.shaman.mvp.presenter.PresenterAbout
import com.chplalex.shaman.mvp.presenter.PresenterSettings
import com.chplalex.shaman.mvp.view.IViewSettings
import com.chplalex.shaman.ui.App
import com.chplalex.shaman.ui.App.Companion.instance
import com.google.android.material.button.MaterialButton
import com.google.android.material.checkbox.MaterialCheckBox
import moxy.MvpAppCompatFragment
import moxy.ktx.moxyPresenter
import javax.inject.Inject
import javax.inject.Provider

class FragmentSettings : MvpAppCompatFragment(R.layout.fragment_settings), IViewSettings {

    @Inject
    lateinit var injectPresenter: Provider<PresenterSettings>

    private val presenter by moxyPresenter {
        injectPresenter.get()
    }

    private lateinit var checkBoxPressure: MaterialCheckBox
    private lateinit var checkBoxWind: MaterialCheckBox
    private lateinit var checkBoxSunMoving: MaterialCheckBox
    private lateinit var checkBoxHumidity: MaterialCheckBox
    private lateinit var rgTheme: RadioGroup
    private lateinit var btnClearPreferences: MaterialButton

    override fun onCreate(savedInstanceState: Bundle?) {
        instance.activityComponent?.inject(this)
        super.onCreate(savedInstanceState)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        activity?.setTitle(R.string.label_settings)
        setHasOptionsMenu(true)
        findViewsById(view)
    }

    private fun findViewsById(view: View) = with(view) {
        checkBoxPressure = findViewById(R.id.checkBoxPressure)
        checkBoxWind = findViewById(R.id.checkBoxWind)
        checkBoxSunMoving = findViewById(R.id.checkBoxSunMoving)
        checkBoxHumidity = findViewById(R.id.checkBoxHumidity)
        rgTheme = findViewById(R.id.rgTheme)
        btnClearPreferences = findViewById(R.id.btnClearPreferences)
    }

    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
        super.onCreateOptionsMenu(menu, inflater)
        inflater.inflate(R.menu.menu_no_start, menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == R.id.action_start) {
            presenter.onActionStart()
            return true
        }
        return super.onOptionsItemSelected(item)
    }

    override fun setListenerHumidity(listener: ((View) -> Unit)?) {
        checkBoxHumidity.setOnClickListener(listener)
    }

    override fun setListenerPressure(listener: ((View) -> Unit)?) {
        checkBoxPressure.setOnClickListener(listener)
    }

    override fun setListenerSunMoving(listener: ((View) -> Unit)?) {
        checkBoxSunMoving.setOnClickListener(listener)
    }

    override fun setListenerWind(listener: ((View) -> Unit)?) {
        checkBoxWind.setOnClickListener(listener)
    }

    override fun setListenerTheme(listener: ((View, Int) -> Unit)?) {
        rgTheme.setOnCheckedChangeListener(listener)
    }

    override fun setListenerClear(listener: ((View) -> Unit)?) {
        btnClearPreferences.setOnClickListener(listener)
    }

    override fun setPressure(isChecked: Boolean) {
        checkBoxPressure.isChecked = isChecked
    }

    override fun setWind(isChecked: Boolean) {
        checkBoxWind.isChecked = isChecked
    }

    override fun setSunMoving(isChecked: Boolean) {
        checkBoxSunMoving.isChecked = isChecked
    }

    override fun setHumidity(isChecked: Boolean) {
        checkBoxHumidity.isChecked = isChecked
    }

    override fun setTheme(id: Int) {
        rgTheme.check(id)
    }

    override fun recreateActivity() {
        activity?.recreate()
    }
}